Crash when using perl

**Reported by wodim on 08 Mar 2011 16:12:55 UTC**

```
panic: MUTEX_LOCK (22) [op.c:476] at (eval 1) line 3.
panic: MUTEX_LOCK (22) [op.c:476].

Program exited with code 0377.
```

It doesn't always crash, it does after using this script several times along a long time (~1 hour?):

```
perl.begin -f

use XML::Simple;
use LWP::Simple;

my $xml = new XML::Simple;
my $content = get("http://ws.audioscrobbler.com/1.0/user/egns/recenttracks.rss");

if(!$content) {
    KVIrc::eval("ECHO Could not retrieve your last tracks.");
} else {
    my $rss = $xml->XMLin($content);
    my $title = $rss->{channel}->{item}[0]->{title};
    KVIrc::eval("ME > $title");
}

perl.end
```

perl 5.12.3-1<br/>kvirc r5553<br/>x86_64

---

Migrated from: https://svn.kvirc.de/kvirc/ticket/1105

--- a/src/modules/perlcore/libkviperlcore.cpp
+++ b/src/modules/perlcore/libkviperlcore.cpp
@@ -435,7 +435,11 @@ static bool perlcore_module_cleanup(KviModule *)
 static bool perlcore_module_can_unload(KviModule *)
 {
 #ifdef COMPILE_PERL_SUPPORT
-	return (g_pInterpreters->count() == 0);
+	return false;
+	/* return (g_pInterpreters->count() == 0);
+	 * Perlcore module can't be cleanly unloaded since PERL_SYS_INIT3 
+	 * and PERL_SYS_TERM should never be called more than once (#1105)
+	 */
 #endif // COMPILE_PERL_SUPPORT
 	return true;
 }

